app-android-xamarin/Moodis/Moodis/Feature/Menu/MenuActivity.cs

```diff
@@ -0,0 +1,155 @@
+        protected override void OnCreate(Bundle savedInstanceState)
+        {
+            base.OnCreate(savedInstanceState);
+            ActivityMenuViewModel = new MenuViewModel();
+            SetContentView(Resource.Layout.activity_menu);
+
+            Intent intent = Intent;
+            ActivityMenuViewModel.currentImage.ImagePath = intent.GetStringExtra("ImagePath");
+
+            InitButtonsAndInputs();
+            UpdateLabels();
+            DeleteImage(ActivityMenuViewModel.currentImage.ImagePath);
+        }
+
+        public async void UpdateLabels()
+        {
+            var imageBox = FindViewById<ImageView>(Resource.Id.imageForView);
+            Bitmap bmImg = BitmapFactory.DecodeFile(ActivityMenuViewModel.currentImage.ImagePath);
+            bmImg = RotateImage(bmImg);
+            imageBox.SetImageBitmap(bmImg);
+            var emotionLabels = new List<TextView> { FindViewById<TextView>(Resource.Id.lblAnger), FindViewById<TextView>(Resource.Id.lblContempt), FindViewById<TextView>(Resource.Id.lblDisgust),
+                FindViewById<TextView>(Resource.Id.lblFear), FindViewById<TextView>(Resource.Id.lblHappiness), FindViewById<TextView>(Resource.Id.lblNeutral), FindViewById<TextView>(Resource.Id.lblSadness),
+                FindViewById<TextView>(Resource.Id.lblSurprise) };
+            foreach (var label in emotionLabels)
+            {
+                label.Text = "loading";
+            }
+            /* COMENTED UNTIL WE FIGURE OUT A WAY TO STORE ENVIROMENTAL VARIABLES
+            try
+            {
+                await ActivityMenuViewModel.GetFaceEmotionsAsync();
+            }
+            catch (System.Net.Http.HttpRequestException e)
+            {
+                Log.Debug(TAG,e.Message);
+                Toast.MakeText(this, WarningInRequest, ToastLength.Short).Show();
+                JavaSystem.Exit(0);
+            }
+            */
+            if (ActivityMenuViewModel.currentImage.emotions != null)
+            {
+                var counter = 0;
+                foreach (var label in emotionLabels)
+                {
+                    label.Text = ActivityMenuViewModel.currentImage.emotions[counter].name + " : "
+                        + ActivityMenuViewModel.currentImage.emotions[counter].confidence.ToString(FormatDouble);
+                    counter++;
+                }
+                ActivityMenuViewModel.UserAddImage();
+            }
+            else
+            {
+                Toast.MakeText(this, WarningFaceDetection, ToastLength.Short).Show();
+            }
+        }
+
+        private Bitmap RotateImage(Bitmap image)
+        {
+            Matrix matrix = new Matrix();
+            matrix.PostRotate(-90);
+            return Bitmap.CreateBitmap(image, 0, 0, image.Width, image.Height, matrix, true);
+        }
+        public override void OnBackPressed()
+        {
+            //Finish(); TODO figure out what to do when back pressed while in menu (maybe logout user ?)
```

Comment:
Few possible ideas would be to either close the app or return to camera activity

Reply:
what about logging user out and taking them back to sign in form ?